src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -256,10 +262,25 @@ private static String getSkippedMessage(Element testCase) {
         return message;
     }
 
-    public String getDisplayName() {
+    public String getTransformedTestName() {
         return TestNameTransformer.getTransformedName(testName);
     }
 
+    public String getDisplayName() {
+        // Only prepend the enclosing flow node names if there are any and the run this is in has multiple blocks directly containing
+        // test results.
+        if (!getEnclosingFlowNodeNames().isEmpty()) {
```

Comment:
> so if we're going that far, hey, why not just stick that all in a transient field and avoid having to do that loading again later

Well, so the *other* option is to cache *all* enclosing blocks for a FlowNode in a List<String>.   So you only do a single lookup, then you can load all the FlowNodes. 

Let's say that in the future I anticipate efficient bulk loading of FlowNodes will become A Thing and leave it at that for now. ;)